R/human_infection.R

```diff
@@ -289,27 +295,54 @@ calculate_treated <- function(
       replace = TRUE
     )
   ])
-
-  successful <- bernoulli_multi_p(parameters$drug_efficacy[drugs])
-  treated_index <- bitset_at(seek_treatment, successful)
-
-  # Update those who have been treated
-  if (treated_index$size() > 0) {
-    variables$state$queue_update('Tr', treated_index)
+  
+  if(!is.null(parameters$antimalarial_resistance)) {
+    antimalarial_resistance_drug_index <- as.numeric(parameters$antimalarial_resistance_drug)[drugs]
+    artemisinin_resistance_proportion <- vector()
+    early_treatment_failure_probability <- vector()
+    drug_indices <- list(); drug_index <- vector()
+    for(i in seq_along(parameters$antimalarial_resistance_drug)) {
+      drug_indices[[i]] <- which(drugs == i)
+      drug_index[i] <- which(parameters$antimalarial_resistance_drug == i)
+      artemisinin_resistance_proportion[drug_indices[[i]]] <- parameters$prop_artemisinin_resistant[[drug_index[i]]][match_timestep(ts = parameters$antimalarial_resistance_timesteps[[drug_index[i]]],
+                                                                                                                                    t = timestep)]
+      early_treatment_failure_probability[drug_indices[[i]]] <- parameters$early_treatment_failure_prob[[drug_index[i]]][match_timestep(ts = parameters$antimalarial_resistance_timesteps[[drug_index[i]]],
+                                                                                                                                        t = timestep)]
```

Comment:
Agree, this is tricky to parse. You call `match_timestep(ts =parameters$antimalarial_resistance_timesteps[[drug_index[i]]],                                                                                                                                t = timestep)` twice with the same parameters, so that can defo come out onto its own line.

I've a sneaking suspicion we can vectorise this whole lot (get rid of the for loop), but might be helpful to simplify first and then see if we can do that.

Reply:
I've shorted these lines by pulling the match_timestep() step out which makes it more readable:

      matched_t <- match_timestep(ts = parameters$antimalarial_resistance_timesteps[[drug_index[i]]], t = timestep)
      artemisinin_resistance_proportion[drug_indices[[i]]] <- parameters$prop_artemisinin_resistant[[drug_index[i]]][matched_t]
      early_treatment_failure_probability[drug_indices[[i]]] <- parameters$early_treatment_failure_prob[[drug_index[i]]][matched_t]

I had another look at vectorising this step, which I agree should be possible, but couldn't puzzle it out yet.